bindings/csharp/Channel.cs

```diff
@@ -69,8 +69,99 @@ public override void write(string str)
             }
         }
 
+        /// <summary><see cref="iio.Channel.ChannelModifier"/> class:
+        /// Contains the available channel modifiers.</summary>
+        public enum ChannelModifier
+        {
+            IIO_NO_MOD,
```

Comment:
Is there a different way to include the C enum? (so we don't need to maintain the list in multiple places) and more important - they are all the same values in the codebase... If someone uses IIO_VOLTAGE, on the C library, and that gets assigned 1, it better be "1" in the C# version as well...

Reply:
A solution to not depend on the values of the enum elements is to create a conversion function between the C type and the C# type (like a lookup table) and of course, the inverse function as well.

Reply:
There is a problem with including the C enum in the C# code. In C the enums are not allocated at an address. They are just interpreted as constants at compile time. This makes them inaccessible in C# via the libiio.dll.
I have discussed with @dNechita and there seem to be two possible solutions for this:
1. Keep a copy of the C enum in the C# code and update it manually every time the C enum in modified. Optionally this process can be automatized with some tools, like Swig.
2. The enums should be placed in a file and CMake should generate their corresponding C header file. If the WITH_CSHARP_BINDINGS flag is set, it should also generate a corresponding C# file containing the enums.

Reply:
Ok - poking around in the [C# doc](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/enums) _An enum declaration that does not explicitly declare an underlying type has an underlying type of int._ That is what I was missing. It might be better to declare things as `public enum ChannelModifier : int`

It might be just easier to modify the existing `CI/travis/check_kernel.sh` (which checks the difference between the latest kernel and the libiio iio.h files) to also check the python and C# bindings.

I think that would resolve things for me - and might be the fastest/easiest.